Include/internal/pycore_interp.h

```diff
@@ -230,6 +237,34 @@ PyAPI_FUNC(int) _PyInterpreterState_IDInitref(PyInterpreterState *);
 PyAPI_FUNC(int) _PyInterpreterState_IDIncref(PyInterpreterState *);
 PyAPI_FUNC(void) _PyInterpreterState_IDDecref(PyInterpreterState *);
 
+#define SIZE_TO_FREELIST_INDEX(size) ((size-4)/2)
```

Comment:
Why divide by 2 and not  `2 * sizeof(void *)`, which is the quantum of allocation for malloc and ob_malloc?
(Assuming that the size is in bytes)
Could you use the term "size class" or equivalent. The mapping here is from sizes to classes (this is important as not all size classes get free lists).

Any reason not to make this an inline function?

Reply:
Alternatively, put an assert in this function, and put a size check in the caller.